src/script/components/InputBar/InputBar.tsx

```diff
@@ -146,83 +138,75 @@ const InputBar = ({
+  const editorRef = useRef<LexicalEditor | null>(null);
+
+  // Typing indicator
   const {typingIndicatorMode} = useKoSubscribableChildren(propertiesRepository, ['typingIndicatorMode']);
   const isTypingIndicatorEnabled = typingIndicatorMode === CONVERSATION_TYPING_INDICATOR_MODE.ON;
-  const shadowInputRef = useRef<HTMLDivElement>(null);
-  const textareaRef = useRef<HTMLTextAreaElement>(null);
 
-  const [editMessageEntity, setEditMessageEntity] = useState<ContentMessage | null>(null);
+  // Message
+  /** the messageContent represents the message being edited. It's directly derived from the editor state */
+  const [messageContent, setMessageContent] = useState<RichTextContent>({text: ''});
+  const [editedMessage, setEditedMessage] = useState<ContentMessage | undefined>();
   const [replyMessageEntity, setReplyMessageEntity] = useState<ContentMessage | null>(null);
-  const [currentMentions, setCurrentMentions] = useState<MentionEntity[]>([]);
+  const textValue = messageContent.text;
+
+  // Files
   const [pastedFile, setPastedFile] = useState<File | null>(null);
-  const [inputValue, setInputValue] = useState<string>('');
-  const [selectionStart, setSelectionStart] = useState<number>(0);
-  const [selectionEnd, setSelectionEnd] = useState<number>(0);
+
+  // Common
   const [pingDisabled, setIsPingDisabled] = useState<boolean>(false);
-  const [isTyping, setIsTyping] = useState<boolean>(false);
-  const hasUserTyped = useRef<boolean>(false);
-  const [editedMention, setEditedMention] = useState<{startIndex: number; term: string} | undefined>(undefined);
 
+  // Right sidebar
   const {rightSidebar} = useAppMainState.getState();
   const lastItem = rightSidebar.history.length - 1;
   const currentState = rightSidebar.history[lastItem];
   const isRightSidebarOpen = !!currentState;
 
   const inputPlaceholder = messageTimer ? t('tooltipConversationEphemeral') : t('tooltipConversationInputPlaceholder');
 
-  const candidates = participatingUserEts.filter(userEntity => !userEntity.isService);
-  const mentionSuggestions = editedMention ? searchRepository.searchUserInSet(editedMention.term, candidates) : [];
-
-  const isEditing = !!editMessageEntity;
+  const isEditing = !!editedMessage;
   const isReplying = !!replyMessageEntity;
   const isConnectionRequest = isOutgoingRequest || isIncomingRequest;
-  const hasLocalEphemeralTimer = isSelfDeletingMessagesEnabled && localMessageTimer && !hasGlobalMessageTimer;
-
-  const richTextInput = getRichTextInput(currentMentions, inputValue);
+  const hasLocalEphemeralTimer = isSelfDeletingMessagesEnabled && !!localMessageTimer && !hasGlobalMessageTimer;
 
   // To be changed when design chooses a breakpoint, the conditional can be integrated to the ui-kit directly
   const isScaledDown = useMatchMedia('max-width: 768px');
 
-  const config = {
-    GIPHY_TEXT_LENGTH: 256,
-  };
+  const showGiphyButton = messageContent.text.length > 0 && messageContent.text.length <= config.GIPHY_TEXT_LENGTH;
```

Comment:
question: can we use `textValue` variable here instead of `messageContent.text`?. It's declared on line `153`.

Reply:
Yes, absolutely 👌 https://github.com/wireapp/wire-webapp/pull/15507/commits/566c8d18dbfc14e800b94ad50719d061d794fd18